create_release.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+# Script to automate release generation for `Plonky3`.
+# 
+# Release generation relies on `release-plz` and `cargo-semver-checks`, where both are used to:
+# - Update the version in `Cargo.toml` files.
+# - Generate a changelog for each sub-crate.
+# - Create a new git tag for this version.
+# - Create a pull request with the changes (Merging PR triggers the CI to publish to `crates.io`).
+
+# Note that most of the logic in this script is generated by models. Generally I don't rely so heavily on this, but we're sort of needing to do all this as a hack to make `release-plz` work, which eventually may get the option to ignore `dev-dependencies`.
+
+# Check if cargo-semver-checks is installed
+check_binary_installed() {
+  local binary_name="$1"
+  if ! type "$binary_name" >/dev/null 2>&1; then
+    echo "$binary_name is not installed."
+    exit 1
+  fi
+}
+
+if [ -z "${GIT_TOKEN}" ]; then
+  echo "\"GIT_TOKEN\" is not set. release-plz will not work without it."
+  exit 1
+fi
+
+current_branch=$(git rev-parse --abbrev-ref HEAD)
+
+if [ "$current_branch" != "main" ] ; then
+  echo "Must be on the main branch to do a release."
+  exit 1
+fi
+
+check_binary_installed "cargo-semver-checks"
+check_binary_installed "release-plz"
+
+# Remove any line that defines nightly-features in a TOML file
```

Comment:
Hm why is this needed, I'd think ideally we'd preserve features so users can choose if they enable nightly stuff?